xbmc/interfaces/AnnouncementManager.h

```diff
@@ -34,17 +34,43 @@ namespace ANNOUNCEMENT
     void AddAnnouncer(IAnnouncer *listener);
     void RemoveAnnouncer(IAnnouncer *listener);
 
-    void Announce(AnnouncementFlag flag, const char *sender, const char *message);
-    void Announce(AnnouncementFlag flag, const char *sender, const char *message, const CVariant &data);
-    void Announce(AnnouncementFlag flag, const char *sender, const char *message,
-        const std::shared_ptr<const CFileItem>& item);
-    void Announce(AnnouncementFlag flag, const char *sender, const char *message,
-        const std::shared_ptr<const CFileItem>& item, const CVariant &data);
+    void Announce(AnnouncementFlag flag, const std::string& message);
+    void Announce(AnnouncementFlag flag, const std::string& message, const CVariant& data);
+    void Announce(AnnouncementFlag flag,
+                  const std::string& message,
+                  const std::shared_ptr<const CFileItem>& item);
+    void Announce(AnnouncementFlag flag,
+                  const std::string& message,
+                  const std::shared_ptr<const CFileItem>& item,
+                  const CVariant& data);
+
+    void Announce(AnnouncementFlag flag, const std::string& sender, const std::string& message);
+    void Announce(AnnouncementFlag flag,
+                  const std::string& sender,
+                  const std::string& message,
+                  const CVariant& data);
+    void Announce(AnnouncementFlag flag,
+                  const std::string& sender,
+                  const std::string& message,
+                  const std::shared_ptr<const CFileItem>& item,
+                  const CVariant& data);
+
+    // The sender is not related to the application name.
+    // Also it's part of Kodi's API - changing it will break
+    // a big number of python addons and third party json consumers.
```

Comment:
considering what else we are doing around python api changes, should we take the opportunity to change this?

Reply:
I think it's not worth it, just to have "kodi" instead of "xbmc" in the json notifications.

Reply:
Thats fine, its not about "kodi" or "xbmc", its more the trademark stuff for rebranding thats been worked on. Im not across all that, so maybe a question for @Darren-Hill or some of the board members who have been dealing with that stuff.
If its going to cause a breaking change, and for trademark purposes it needs to be changed, probably best to do with our already existing breaking changes. if its not deemed an issue, sorry for the noise

Reply:
Afaik it's not needed for trademark purposes, otherwise it should have been part of the kodi rebrand already - it's not really user facing, just some internal string that is sent to a socket, could be "asdf" as well - I bet most json clients don't care about it at all, at least I don't really understand why it's there at all.
Normally if I as a client connect to a ip/socket pair, I already know which service I connect to, so why would I need a "sender" identifier?
Only one kodi/xbmc/rebranded-fork can bind that socket anyway.

Reply:
A similar question has actually come up as part of the trademark/rebranding work, as for binary addons if you rebrand and change the app name the binary addons do still get installed in a Kodi directory rather than using the new rebrand name. @ronie did some work on this and that was the only issue that got identified from it (via changing APP_NAME in version.txt). 

See his comment on 16th Dec in [this issue](https://github.com/xbmc/board/issues/118) for more details of what he found.

As noted it is questionable whether usage of xbmc/Kodi within the code itself it subject to trademark or not (or indeed if such uses can even be trademarked), but the above is a known blocking point to making rebranding easier for those users who want to go that way.

Reply:
Yeah, while above is probably a bigger issue for rebranders (so is that you need to change include paths in each and every binary addon you want to use), the question in discussion is 
neglectable from a rebranders POV - we could simply add it to the rebrand guide that you can change this new constant if you don't want to have the term "xbmc" in your rebrands json notifications -> probably remotes like KORE etc. will not work with your rebrand then.

The question in matter is:
Was it an oversight, that we as team kodi, need to rebrand this string, or is it save to leave it as is.